ament_index_python/Quality_Declaration.md

```diff
@@ -0,0 +1,185 @@
+
+
+This document is a declaration of software quality for the `ament_index_python` package, based on the guidelines in [REP-2004](https://github.com/ros-infrastructure/rep/blob/rep-2004/rep-2004.rst).
+
+# `ament_index_python` Quality Declaration
+
+The package `ament_index_python` claims to be in the **Quality Level 4** category.
+
+Below are the rationales, notes, and caveats for this claim, organized by each requirement listed in the [Package Quality Categories in REP-2004](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#package-quality-categories)
+
+## Version Policy [1]
+
+### Version Scheme [1.i]
+
+`ament_index_python` uses `semver` according to the recommendation for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#versioning)
+
+### Version Stability [1.ii]
+
+Currently this package version is 0.8.0.''
+
+### Public API Declaration [1.iii]
+
+Not defined API for this package
```

Comment:
I thought this was meant to be declared along with the "Features" documentation, stating it in the QD document will be valid IMO. However, what about the `cli.py` module? This is accessible as a package script

Reply:
Going off one of the earlier examples, [rcutils](https://github.com/ros2/rcutils/blob/master/QUALITY_DECLARATION.md#public-api-declaration-1iii), the public API is kinda just declared in the QD. While the API is in the doxygen documentation, it's never explicitly stated anywhere else what the rcutils API is (as far as I can find). 

I guess I figured that was the point of section 1.iii, but maybe it's supposed be declared somewhere else?

Anything else that is intended to be used by packages outside of this package can probably be considered public API.

Reply:
The case of [rcutils](https://github.com/ros2/rcutils/blob/master/QUALITY_DECLARATION.md#public-api-declaration-1iii) is slightly different because `rcutils` is a C/C++ package, and for those the public headers are considered a valid definition of public API (according to [REP-2004](https://github.com/ros-infrastructure/rep/pull/218/files#diff-f40b01badf3f268cbb5d72ffcc9613f5R494), require a bit extra attention)

I think mentioning the API in a possible [README](./Readme.md) of this package will be enough to consider it a "declared API". Let me know your thoughts.

I have mixed feelings with the `cli` module as well, because their functions are not available but the script application is. I think it shouldn't be considered public API, and in that case I can add to the README file the rest of the functions.

Reply:
@dirk-thomas Can I have your thoughts on this as maintainer of the package?

Reply:
I am sorry, but I don't have time for this atm.